diff -urN nted-1.10.18/chordorrest.cpp nted-1.10.18-patched/chordorrest.cpp
--- nted-1.10.18/chordorrest.cpp	2011-01-27 23:09:44.000000000 +1000
+++ nted-1.10.18-patched/chordorrest.cpp	2014-01-27 19:42:03.000000000 +1100
@@ -3794,7 +3794,7 @@
 					strcat(Str, " ");
 					len2 = NedResource::getPartLength(&dotcount);
 				}
-				fprintf(fp, Str);
+				fprintf(fp, "%s", Str);
 				*midi_len = 0;
 			}
 			else {
@@ -3839,7 +3839,7 @@
 				force_length = true;
 			}
 			if (force_length) {
-				fprintf(fp, Str);
+				fprintf(fp, "%s", Str);
 			}
 			exportLilyAccents(fp);
 			if (keep_beams && m_beam) {
@@ -4153,7 +4153,7 @@
 			strcat(Str, " ");
 			len2 = NedResource::getPartLength(&dotcount);
 		}
-		fprintf(fp, Str);
+		fprintf(fp, "%s", Str);
 		*midi_len = 0;
 	}
 	else {
@@ -4210,7 +4210,7 @@
 			strcat(Str, " ");
 			len2 = NedResource::getPartLength(&dotcount);
 		}
-		fprintf(fp, Str);
+		fprintf(fp, "%s", Str);
 		*midi_len = 0;
 	}
 	else {
diff -urN nted-1.10.18/chords/chordpainter.cpp nted-1.10.18-patched/chords/chordpainter.cpp
--- nted-1.10.18/chords/chordpainter.cpp	2011-01-27 23:09:28.000000000 +1000
+++ nted-1.10.18-patched/chords/chordpainter.cpp	2014-01-27 19:38:26.000000000 +1100
@@ -461,7 +461,7 @@
 		for (i = 0; i < dotcount; i++) {
 			strcat(Str, ".");
 		}
-		fprintf(fp, Str);
+		fprintf(fp, "%s", Str);
 	}
 	*midi_len = duration;
 	if (mody_name[0] != '\0') {
diff -urN nted-1.10.18/mainwindow.cpp nted-1.10.18-patched/mainwindow.cpp
--- nted-1.10.18/mainwindow.cpp	2011-01-27 23:09:44.000000000 +1000
+++ nted-1.10.18-patched/mainwindow.cpp	2014-01-27 19:42:41.000000000 +1100
@@ -3902,8 +3902,8 @@
 		if (m_staff_contexts[i].m_staff_short_name != NULL && strlen(m_staff_contexts[i].m_staff_short_name->getText()) > 0) {
 			fprintf(fp, " \\set Staff.shortInstrumentName = \"%s \"", m_staff_contexts[i].m_staff_short_name->getText());
 		}
-		fprintf(fp, NedResource::getLilyPondClefName(m_staff_contexts[i].m_clef_number));
-		fprintf(fp, NedResource::getLilyPondKeySigName(m_staff_contexts[i].m_key_signature_number));
+		fprintf(fp, "%s", NedResource::getLilyPondClefName(m_staff_contexts[i].m_clef_number));
+		fprintf(fp, "%s", NedResource::getLilyPondKeySigName(m_staff_contexts[i].m_key_signature_number));
 		fprintf(fp, " \\time %d/%d", m_numerator, m_denominator);
 		if (m_upbeat_inverse != 0) {
 			fprintf(fp, " \\partial 64*%d", partial);
diff -urN nted-1.10.18/resource.cpp nted-1.10.18-patched/resource.cpp
--- nted-1.10.18/resource.cpp	2011-01-27 23:09:44.000000000 +1000
+++ nted-1.10.18-patched/resource.cpp	2014-01-27 19:43:10.000000000 +1100
@@ -3930,7 +3930,7 @@
 		for (i = 0; i < dotcount; i++) {
 			strcat(Str, ".");
 		}
-		fprintf(fp, Str);
+		fprintf(fp, "%s", Str);
 	}
 	*midi_len = duration;
 	if (chordname->getUpText() != '\0') {
