--- nted-1.9.18/chordorrest.cpp.str	2010-02-03 12:17:32.000000000 +0100
+++ nted-1.9.18/chordorrest.cpp	2010-02-03 12:18:23.000000000 +0100
@@ -3672,7 +3672,7 @@
 					strcat(Str, " ");
 					len2 = NedResource::getPartLength(&dotcount);
 				}
-				fprintf(fp, Str);
+				fprintf(fp, "%s", Str);
 				*midi_len = 0;
 			}
 			else {
@@ -3706,7 +3706,7 @@
 				force_length = true;
 			}
 			if (force_length) {
-				fprintf(fp, Str);
+				fprintf(fp, "%s", Str);
 			}
 			exportLilyAccents(fp);
 			if (keep_beams && m_beam) {
@@ -4013,7 +4013,7 @@
 			strcat(Str, " ");
 			len2 = NedResource::getPartLength(&dotcount);
 		}
-		fprintf(fp, Str);
+		fprintf(fp, "%s", Str);
 		*midi_len = 0;
 	}
 	else {
@@ -4070,7 +4070,7 @@
 			strcat(Str, " ");
 			len2 = NedResource::getPartLength(&dotcount);
 		}
-		fprintf(fp, Str);
+		fprintf(fp, "%s", Str);
 		*midi_len = 0;
 	}
 	else {
--- nted-1.9.18/chords/chordpainter.cpp.str	2010-02-03 12:15:40.000000000 +0100
+++ nted-1.9.18/chords/chordpainter.cpp	2010-02-03 12:15:51.000000000 +0100
@@ -461,7 +461,7 @@
 		for (i = 0; i < dotcount; i++) {
 			strcat(Str, ".");
 		}
-		fprintf(fp, Str);
+		fprintf(fp, "%s", Str);
 	}
 	*midi_len = duration;
 	if (mody_name[0] != '\0') {
--- nted-1.9.18/mainwindow.cpp.str	2010-02-03 12:16:09.000000000 +0100
+++ nted-1.9.18/mainwindow.cpp	2010-02-03 12:16:29.000000000 +0100
@@ -3832,8 +3832,8 @@
 		if (m_staff_contexts[i].m_staff_short_name != NULL && strlen(m_staff_contexts[i].m_staff_short_name->getText()) > 0) {
 			fprintf(fp, " \\set Staff.shortInstrumentName = \"%s \"", m_staff_contexts[i].m_staff_short_name->getText());
 		}
-		fprintf(fp, NedResource::getLilyPondClefName(m_staff_contexts[i].m_clef_number));
-		fprintf(fp, NedResource::getLilyPondKeySigName(m_staff_contexts[i].m_key_signature_number));
+		fprintf(fp, "%s", NedResource::getLilyPondClefName(m_staff_contexts[i].m_clef_number));
+		fprintf(fp, "%s", NedResource::getLilyPondKeySigName(m_staff_contexts[i].m_key_signature_number));
 		fprintf(fp, " \\time %d/%d", m_numerator, m_denominator);
 		if (m_upbeat_inverse != 0) {
 			fprintf(fp, " \\partial 64*%d", partial);
--- nted-1.9.18/resource.cpp.str	2010-02-03 12:16:51.000000000 +0100
+++ nted-1.9.18/resource.cpp	2010-02-03 12:17:01.000000000 +0100
@@ -3783,7 +3783,7 @@
 		for (i = 0; i < dotcount; i++) {
 			strcat(Str, ".");
 		}
-		fprintf(fp, Str);
+		fprintf(fp, "%s", Str);
 	}
 	*midi_len = duration;
 	if (chordname->getUpText() != '\0') {
